scripts/puritan.lic

```diff
@@ -313,28 +313,41 @@ module Puritan
     end
   end
 
-  def self.outcome_of_1004(gem)
-    waitcastrt?
-    waitrt?
-    Spell.lock_cast()
-    prep_1004()
-    fput("sing ##{gem.id}")
-    if wait_for_own_song().eql?(:ok)
-      result = matchwait(
-        #"gem becomes more perfect",
-        #"improves somewhat",
+  def self.select_outcome_from_1004_output()
```

Comment:
This logic does what I want but it feels super awkward. I don't usually have this problem in Ruby but my brain is not cooperating today.

Reply:
consuming these pseudo `IO` objects from Lich is always awkward because the don't implement the proper `IO` interfaces.  I've written a half dozen APIs around this problem and never been happy myself yet.